commerce-payment-method-mercanet/src/main/java/com/liferay/commerce/payment/method/mercanet/internal/MercanetCommercePaymentMethod.java

```diff
@@ -193,12 +221,10 @@ public CommercePaymentResult processPayment(
 		paymentRequest.setOrderId(
 			String.valueOf(commerceOrder.getCommerceOrderId()));
 
-		StringBundler transactionReferenceSB = new StringBundler(5);
+		StringBundler transactionReferenceSB = new StringBundler(3);
 
 		transactionReferenceSB.append(commerceOrder.getCompanyId());
-		transactionReferenceSB.append(StringPool.POUND);
```

Comment:
Hey @luca-pellizzon, is there a specific reason we have to exclude these delimiters? If there's no delimiters then there's a chance the result is not unique -- e.g.

    companyId=1, groupId=234, orderId=567 => 1234567
    companyId=1, groupId=23, orderId=4567 => 1234567

Do you know what would happen if we had a duplicate?

Reply:
Mercanet does not accept any "special" character.
We need to find an unique combination. We can't use the uuid because it is too long.